### PR TITLE
Fix semver ordering for libraries

### LINKED ISSUE #5085: [BUG]: Sorting libraries based on semver doesn't actually work
### Describe the bug

#4153 claimed to sort libraries based on semver (fixing #4142), but it doesn't seem to be completely right.

For instance, `mp-units` is ordered oldest-to-newest with trunk last:
* 0.3.1
* 0.4.0
* ...
* 0.7.0
* trunk

`fmt` is ordered newest-to-oldest with trunk first:
* trunk
* 9.1.0
* 9.0.0
* ...
* 4.1.0
* 4.0.0

`GLM` is ordered newest-to-oldest, except with trunk last:
* 0.9.9.8
* 0.9.9.7
* ...
* 0.9.8.5
* trunk

Also that PR updated `options-handler.js` but not `options-handler.ts`, which also has this comment which seems relevant: https://github.com/compiler-explorer/compiler-explorer/blob/0364ef87ad28e5176b62d2b25f13fb42579baf6c/lib/options-handler.ts#L365-L369

So something isn't quite right here. 

### Steps to reproduce

Look at the version ordering for `mp-units` vs `fmt` vs `GLM` in C++.

### Expected behavior

The expected behavior is that `trunk` should go first (if present) and then all the semver versions in decreasing order. 

### Reproduction link

Not applicable

### Screenshots

Not applicable

### Operating System

_No response_

### Browser version

_No response_

### LINKED ISSUE #5951: [BUG]: Library version ordering
There's a TODO in the code for library version sorting:
https://github.com/compiler-explorer/compiler-explorer/blame/d81101336384a3e8d507076d5bf43cf95d844716/lib/options-handler.ts#L365

Most libraries seem to still have descending semver ordering, but not all

This is because it's naturally ordered in how they appear in the version list in the properties file and then reversed, and people usually put the latest version last, which then works out as intended.

Would be good to have actual semver ordering, I think the correct property should be `.version` instead of `.semver`
https://github.com/compiler-explorer/compiler-explorer/blob/main/lib/options-handler.ts#L303

Libraries versions that are 'trunk' or 'main' should probably be on top like they are now.

### PR DESCRIPTION
**Pull request by partouf**
Fix #5951
Fix #5085

Extended the fictitious version number for non-numbers, so that trunk can be listed higher than a `yyyymmdd` as a version (EVE uses this)

### CODE REVIEW

File: lib/utils.ts [hunk 1]
```diff
@@ -516,6 +516,10 @@ export function asSafeVer(semver: string | number | null | undefined) {
                 return validated;
             }
         }
+
+        if (semver.includes('trunk') || semver.includes('main')) {
+            return '99999999.99999.999';
```
Comment (mattgodbolt) [inline lib/utils.ts:521]: bit of a hack but I hve no better idea :) Mayyybe we should make a `const` value for the `99999.99.99` constant though?

Comment (mattgodbolt) [reply lib/utils.ts:521]: Ohhhh! scratch that! I didn't sot they were actualy different...

So maybe...name them like you've done in the tests? Then old eyes like mine might realise they're different and it's on purpose :)

**Review by mattgodbolt — APPROVED** [review]

File: lib/utils.ts (hunk 1 above)
Comment (partouf) [reply lib/utils.ts:521]: yeah that's what i was thinking as well

Comment (partouf) [reply lib/utils.ts]: fixed

**Pull request merged**